_examples/01_one_account_one_region/README.md

```diff
@@ -0,0 +1,155 @@
+      versioned, which allows each environment in the infrastructure to use a different
+      version of the Terraform module. This allows for straightforward development and
+      testing of Terraform module changes in QA and staging environments before applying
+      those changes to production.
+     
+      In this example, look at the `source` stanza of the `terraform` block in any of the
+      component configuration files. Notice that each specifies a specific version of the
+      Terraform module source by using the `?ref=<some version>` syntax.
+2. Independent remote state
+   1. Typically, a Terraform project will have a single top level `main.tf` file
+      for each environment, which calls a Terraform module for each component of
+      the infrastructure. As a result of having a single top level `main.tf` file,
+      the entire environment has a single remote state file.
+   
+      Hosting each Terraform module in its own repo makes it possible to setup the directory
+      structure such that each component within an environment has its own remote state file.
+      In this example, each component is defined in its own directory. This means that when
+      Terraform is run in each directory, a remote state file will be created for only the
+      infrastructure defined within that directory; a single component within the environment.
+      
+      Having a remote state file for each component makes it easier to make changes to a single
+      component; commands like `plan` and `apply` will be quicker because they only have to refresh
+      the state of the single component, not the entire environment. Also, each state file is smaller
+      in scope, which means that we have a smaller blast radius if a bug is introduced into one of
+      our components. It is more likely to break a single component instead of the entire environment.
+      
+## Dependencies between infrastructure components
+It might be best to come up with some contrived example since this is the first one and *NOT* introduce
+the `dependencies` keyword yet. Save it for a future example so that we can explain it appropriately?
+
+Some things to discuss with the `dependencies` keyword:
+* makes `apply-all` and `destroy-all` work as expected
+* issues with `plan-all`
+
+If we do leave this in the first example, this section would also explain that the dependency
+between the two modules is because the
+[conorgil/terragrunt_sample_frontend_app](https://github.com/conorgil/terragrunt_sample_frontend_app)
+Terraofrm module needs to pull from the 
+[conorgil/terragrunt_sample_vpc](https://github.com/conorgil/terragrunt_sample_vpc)
+module which subnet to use for the EC2 instance. Highlight that the 
+[conorgil/terragrunt_sample_frontend_app](https://github.com/conorgil/terragrunt_sample_frontend_app)
+module accomplishes this by using a `terraform_remote_state` data source and exposes the configuration

```

Comment:
That is a super interesting idea. I literally did not even think about using data sources instead of `terraform_remote_state`.

The team that I was working with started using Terraform a few years ago before data sources were introduced and we never updated our configurations to use them because of the classic "everything works and we'll update it later" logic.

I agree that both approaches are functionally equivalent and come down to user preference. I will leave them as `terraform_remote_state` for now since that is what I am familiar with.